File: backend/models/inventory.py
"""
Inventory Model - Advanced product and stock management
"""
import sqlite3
import json
import qrcode
import io
import base64
from datetime import datetime
from typing import Dict, List, Optional, Tuple
from decimal import Decimal, ROUND_HALF_UP

class InventoryModel:
    def __init__(self, db_path: str = "pos_database.db"):
        self.db_path = db_path
        self.init_database()
    
    def init_database(self):
        """Initialize inventory tables"""
        conn = sqlite3.connect(self.db_path)
        cursor = conn.cursor()
        
        # Products table
        cursor.execute('''
            CREATE TABLE IF NOT EXISTS products (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                name TEXT NOT NULL,
                description TEXT,
                sku TEXT UNIQUE NOT NULL,
                barcode TEXT UNIQUE,
                category_id INTEGER,
                price DECIMAL(10,2) NOT NULL,
                cost DECIMAL(10,2),
                stock_quantity INTEGER DEFAULT 0,
                min_stock_level INTEGER DEFAULT 5,
                max_stock_level INTEGER DEFAULT 100,
                unit TEXT DEFAULT 'each',
                weight DECIMAL(8,3),
                dimensions TEXT,
                supplier_id INTEGER,
                is_active BOOLEAN DEFAULT 1,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
                updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
                FOREIGN KEY (category_id) REFERENCES categories(id),
                FOREIGN KEY (supplier_id) REFERENCES suppliers(id)
            )
        ''')
        
        # Categories table
        cursor.execute('''
            CREATE TABLE IF NOT EXISTS categories (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                name TEXT NOT NULL,
                description TEXT,
                parent_id INTEGER,
                color_code TEXT DEFAULT '#00D4FF',
                is_active BOOLEAN DEFAULT 1,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
                FOREIGN KEY (parent_id) REFERENCES categories(id)
            )
        ''')
        
        # Stock movements table
        cursor.execute('''
            CREATE TABLE IF NOT EXISTS stock_movements (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                product_id INTEGER NOT NULL,
                movement_type TEXT NOT NULL,
                quantity INTEGER NOT NULL,
                previous_quantity INTEGER NOT NULL,
                new_quantity INTEGER NOT NULL,
                reason TEXT,
                reference_id INTEGER,
                employee_id INTEGER,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
                FOREIGN KEY (product_id) REFERENCES products(id),
                FOREIGN KEY (employee_id) REFERENCES employees(id)
            )
        ''')
        
        # Suppliers table
        cursor.execute('''
            CREATE TABLE IF NOT EXISTS suppliers (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                name TEXT NOT NULL,
                contact_person TEXT,
                email TEXT,
                phone TEXT,
                address TEXT,
                is_active BOOLEAN DEFAULT 1,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
            )
        ''')
        
        # Insert default categories
        default_categories = [
            ('Electronics', 'Electronic devices and accessories', None, '#00D4FF'),
            ('Clothing', 'Apparel and fashion items', None, '#00FFAB'),
            ('Food & Beverage', 'Food and drink items', None, '#FFD700'),
            ('Accessories', 'Miscellaneous accessories', None, '#E8E8E8')
        ]
        
        cursor.executemany('''
            INSERT OR IGNORE INTO categories (name, description, parent_id, color_code) 
            VALUES (?, ?, ?, ?)
        ''', default_categories)
        
        conn.commit()
        conn.close()
    
    def add_product(self, product_data: Dict) -> Dict:
        """Add a new product to inventory"""
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            
            # Generate SKU if not provided
            if not product_data.get('sku'):
                product_data['sku'] = self._generate_sku(product_data['name'])
            
            # Generate barcode if not provided
            if not product_data.get('barcode'):
                product_data['barcode'] = self._generate_barcode()
            
            cursor.execute('''
                INSERT INTO products (
                    name, description, sku, barcode, category_id, price, cost,
                    stock_quantity, min_stock_level, max_stock_level, unit,
                    weight, dimensions, supplier_id
                ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
            ''', (
                product_data['name'],
                product_data.get('description', ''),
                product_data['sku'],
                product_data['barcode'],
                product_data.get('category_id'),
                float(product_data['price']),
                float(product_data.get('cost', 0)),
                int(product_data.get('stock_quantity', 0)),
                int(product_data.get('min_stock_level', 5)),
                int(product_data.get('max_stock_level', 100)),
                product_data.get('unit', 'each'),
                float(product_data.get('weight', 0)),
                product_data.get('dimensions', ''),
                product_data.get('supplier_id')
            ))
            
            product_id = cursor.lastrowid
            
            # Record initial stock movement
            if product_data.get('stock_quantity', 0) > 0:
                cursor.execute('''
                    INSERT INTO stock_movements (
                        product_id, movement_type, quantity, previous_quantity,
                        new_quantity, reason, employee_id
                    ) VALUES (?, ?, ?, ?, ?, ?, ?)
                ''', (
                    product_id, 'initial_stock', product_data['stock_quantity'],
                    0, product_data['stock_quantity'], 'Initial stock entry',
                    product_data.get('employee_id')
                ))
            
            conn.commit()
            conn.close()
            
            return {
                'success': True,
                'product_id': product_id,
                'sku': product_data['sku'],
                'barcode': product_data['barcode']
            }
            
        except Exception as e:
            return {'success': False, 'error': str(e)}
    
    def update_product(self, product_id: int, product_data: Dict) -> Dict:
        """Update existing product"""
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            
            # Build dynamic update query
            update_fields = []
            values = []
            
            for field in ['name', 'description', 'price', 'cost', 'min_stock_level', 
                         'max_stock_level', 'unit', 'weight', 'dimensions', 'supplier_id']:
                if field in product_data:
                    update_fields.append(f"{field} = ?")
                    values.append(product_data[field])
            
            if update_fields:
                values.append(product_id)
                cursor.execute(f'''
                    UPDATE products 
                    SET {', '.join(update_fields)}, updated_at = CURRENT_TIMESTAMP
                    WHERE id = ?
                ''', values)
            
            conn.commit()
            conn.close()
            
            return {'success': True, 'message': 'Product updated successfully'}
            
        except Exception as e:
            return {'success': False, 'error': str(e)}
    
    def delete_product(self, product_id: int) -> Dict:
        """Soft delete product (mark as inactive)"""
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            
            cursor.execute('''
                UPDATE products 
                SET is_active = 0, updated_at = CURRENT_TIMESTAMP
                WHERE id = ?
            ''', (product_id,))
            
            conn.commit()
            conn.close()
            
            return {'success': True, 'message': 'Product deleted successfully'}
            
        except Exception as e:
            return {'success': False, 'error': str(e)}
    
    def get_products(self, category_id: Optional[int] = None, 
                    search_term: Optional[str] = None, 
                    low_stock_only: bool = False) -> Dict:
        """Get products with optional filtering"""
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            
            query = '''
                SELECT p.*, c.name as category_name, s.name as supplier_name
                FROM products p
                LEFT JOIN categories c ON p.category_id = c.id
                LEFT JOIN suppliers s ON p.supplier_id = s.id
                WHERE p.is_active = 1
            '''
            
            params = []
            
            if category_id:
                query += ' AND p.category_id = ?'
                params.append(category_id)
            
            if search_term:
                query += ' AND (p.name LIKE ? OR p.sku LIKE ? OR p.barcode LIKE ?)'
                search_pattern = f'%{search_term}%'
                params.extend([search_pattern, search_pattern, search_pattern])
            
            if low_stock_only:
                query += ' AND p.stock_quantity <= p.min_stock_level'
            
            query += ' ORDER BY p.name'
            
            cursor.execute(query, params)
            products = cursor.fetchall()
            
            # Convert to list of dictionaries
            product_list = []
            for product in products:
                product_list.append({
                    'id': product[0],
                    'name': product[1],
                    'description': product[2],
                    'sku': product[3],
                    'barcode': product[4],
                    'category_id': product[5],
                    'category_name': product[16],
                    'price': float(product[6]),
                    'cost': float(product[7]) if product[7] else 0,
                    'stock_quantity': product[8],
                    'min_stock_level': product[9],
                    'max_stock_level': product[10],
                    'unit': product[11],
                    'weight': float(product[12]) if product[12] else 0,
                    'dimensions': product[13],
                    'supplier_id': product[14],
                    'supplier_name': product[17],
                    'is_active': bool(product[15]),
                    'created_at': product[18],
                    'updated_at': product[19]
                })
            
            conn.close()
            
            return {'success': True, 'products': product_list}
            
        except Exception as e:
            return {'success': False, 'error': str(e)}
    
    def update_stock(self, product_id: int, quantity_change: int, 
                    movement_type: str, reason: str = "", 
                    employee_id: Optional[int] = None) -> Dict:
        """Update product stock quantity"""
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            
            # Get current stock
            cursor.execute('SELECT stock_quantity FROM products WHERE id = ?', (product_id,))
            current_stock = cursor.fetchone()
            
            if not current_stock:
                return {'success': False, 'error': 'Product not found'}
            
            previous_quantity = current_stock[0]
            new_quantity = previous_quantity + quantity_change
            
            # Prevent negative stock
            if new_quantity < 0:
                return {'success': False, 'error': 'Insufficient stock'}
            
            # Update stock quantity
            cursor.execute('''
                UPDATE products 
                SET stock_quantity = ?, updated_at = CURRENT_TIMESTAMP
                WHERE id = ?
            ''', (new_quantity, product_id))
            
            # Record stock movement
            cursor.execute('''
                INSERT INTO stock_movements (
                    product_id, movement_type, quantity, previous_quantity,
                    new_quantity, reason, employee_id
                ) VALUES (?, ?, ?, ?, ?, ?, ?)
            ''', (product_id, movement_type, quantity_change, previous_quantity, 
                  new_quantity, reason, employee_id))
            
            conn.commit()
            conn.close()
            
            return {
                'success': True,
                'previous_quantity': previous_quantity,
                'new_quantity': new_quantity,
                'quantity_change': quantity_change
            }
            
        except Exception as e:
            return {'success': False, 'error': str(e)}
    
    def get_low_stock_alerts(self) -> Dict:
        """Get products with low stock levels"""
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            
            cursor.execute('''
                SELECT p.*, c.name as category_name
                FROM products p
                LEFT JOIN categories c ON p.category_id = c.id
                WHERE p.is_active = 1 AND p.stock_quantity <= p.min_stock_level
                ORDER BY (p.stock_quantity - p.min_stock_level) ASC
            ''')
            
            low_stock_products = cursor.fetchall()
            
            # Convert to list of dictionaries
            alert_list = []
            for product in low_stock_products:
                alert_list.append({
                    'id': product[0],
                    'name': product[1],
                    'sku': product[3],
                    'current_stock': product[8],
                    'min_stock_level': product[9],
                    'category_name': product[16],
                    'days_remaining': self._calculate_days_remaining(product[0])
                })
            
            conn.close()
            
            return {'success': True, 'alerts': alert_list}
            
        except Exception as e:
            return {'success': False, 'error': str(e)}
    
    def generate_barcode(self, product_id: int) -> Dict:
        """Generate barcode for product"""
        try:
            conn = sqlite3.connect(self.db_path)
            cursor = conn.cursor()
            
            cursor.execute('SELECT barcode, name FROM products WHERE id = ?', (product_id,))
            product = cursor.fetchone()
            
            if not product:
                return {'success': False, 'error': 'Product not found'}
            
            barcode_data = product[0]
            product_name = product[1]
            
            # Generate QR code
            qr = qrcode.QRCode(version=1, box_size=10, border=5)
            qr.add_data(barcode_data)
            qr.make(fit=True)
            
            # Create image
            img = qr.make_image(fill_color="black", back_color="white")
            
            # Convert to base64
            buffer = io.BytesIO()
            img.save(buffer, format='PNG')
            img_str = base64.b64encode(buffer.getvalue()).decode()
            
            conn.close()
            
            return {
                'success': True,
                'barcode_data': barcode_data,
                'barcode_image': f'data:image/png;base64,{img_str}',
                'product_name': product_name
            }
            
        except Exception as e:
            return {'success': False, 'error': str(e)}
    
    def search_products(self, search_term: str) -> Dict:
        """Search products by name, SKU, or barcode"""
        return self.get_products(search_term=search_term)
    
    def _generate_sku(self, product_name: str) -> str:
        """Generate SKU from product name"""
        import re
        # Take first 3 letters of each word, uppercase
        words = re.findall(r'\w+', product_name.upper())
        sku_base = ''.join([word[:3] for word in words[:3]])
        
        # Add timestamp suffix
        timestamp = datetime.now().strftime("%m%d%H%M")
        return f"{sku_base}{timestamp}"
    
    def _generate_barcode(self) -> str:
        """Generate unique barcode"""
        import random
        # Generate 12-digit barcode
        barcode = ''.join([str(random.randint(0, 9)) for _ in range(12)])
        return barcode
    
    def _calculate_days_remaining(self, product_id: int) -> int:
        """Calculate estimated days until stock runs out"""
        # This would analyze historical sales data
        # For now, return a placeholder
        return 7

File: backend/routes/inventory.js
const express = require('express');
const router = express.Router();

// Import models
const InventoryModel = require('../models/inventory');
const BarcodeGenerator = require('../utils/barcode_maker');
const AuthenticationManager = require('../utils/authentication');

// Initialize models
const inventoryModel = new InventoryModel();
const barcodeGenerator = new BarcodeGenerator();
const authManager = new AuthenticationManager();

// Middleware to verify authentication
const authenticateUser = async (req, res, next) => {
  try {
    const session_token = req.cookies.session_token || req.headers.authorization?.replace('Bearer ', '');
    
    if (!session_token) {
      return res.status(401).json({
        success: false,
        message: 'Authentication required'
      });
    }

    const session = await authManager.validate_session(session_token);
    if (!session) {
      return res.status(401).json({
        success: false,
        message: 'Invalid or expired session'
      });
    }

    req.user = session;
    next();
  } catch (error) {
    res.status(500).json({
      success: false,
      message: 'Authentication error'
    });
  }
};

// Get all products
router.get('/products', authenticateUser, async (req, res) => {
  try {
    const { category_id, search, low_stock_only } = req.query;

    const result = await inventoryModel.get_products(
      category_id ? parseInt(category_id) : null,
      search || null,
      low_stock_only === 'true'
    );

    if (result.success) {
      res.json({
        success: true,
        data: result.products
      });
    } else {
      res.status(500).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Get products error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to get products'
    });
  }
});

// Get product by ID
router.get('/products/:id', authenticateUser, async (req, res) => {
  try {
    const { id } = req.params;

    const result = await inventoryModel.get_products();
    
    if (result.success) {
      const product = result.products.find(p => p.id === parseInt(id));
      
      if (product) {
        res.json({
          success: true,
          data: product
        });
      } else {
        res.status(404).json({
          success: false,
          message: 'Product not found'
        });
      }
    } else {
      res.status(500).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Get product error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to get product'
    });
  }
});

// Add new product
router.post('/products', authenticateUser, async (req, res) => {
  try {
    // Check permissions
    if (!req.user.permissions?.inventory?.includes('create')) {
      return res.status(403).json({
        success: false,
        message: 'Permission denied: Cannot create products'
      });
    }

    const product_data = {
      ...req.body,
      employee_id: req.user.user_id
    };

    // Validate required fields
    if (!product_data.name || !product_data.price) {
      return res.status(400).json({
        success: false,
        message: 'Product name and price are required'
      });
    }

    const result = await inventoryModel.add_product(product_data);

    if (result.success) {
      res.status(201).json({
        success: true,
        message: 'Product created successfully',
        data: {
          product_id: result.product_id,
          sku: result.sku,
          barcode: result.barcode
        }
      });
    } else {
      res.status(400).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Add product error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to add product'
    });
  }
});

// Update product
router.put('/products/:id', authenticateUser, async (req, res) => {
  try {
    // Check permissions
    if (!req.user.permissions?.inventory?.includes('update')) {
      return res.status(403).json({
        success: false,
        message: 'Permission denied: Cannot update products'
      });
    }

    const { id } = req.params;
    const product_data = req.body;

    const result = await inventoryModel.update_product(parseInt(id), product_data);

    if (result.success) {
      res.json({
        success: true,
        message: 'Product updated successfully'
      });
    } else {
      res.status(400).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Update product error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to update product'
    });
  }
});

// Delete product
router.delete('/products/:id', authenticateUser, async (req, res) => {
  try {
    // Check permissions
    if (!req.user.permissions?.inventory?.includes('delete')) {
      return res.status(403).json({
        success: false,
        message: 'Permission denied: Cannot delete products'
      });
    }

    const { id } = req.params;

    const result = await inventoryModel.delete_product(parseInt(id));

    if (result.success) {
      res.json({
        success: true,
        message: 'Product deleted successfully'
      });
    } else {
      res.status(400).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Delete product error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to delete product'
    });
  }
});

// Update stock
router.post('/products/:id/stock', authenticateUser, async (req, res) => {
  try {
    // Check permissions
    if (!req.user.permissions?.inventory?.includes('update')) {
      return res.status(403).json({
        success: false,
        message: 'Permission denied: Cannot update inventory'
      });
    }

    const { id } = req.params;
    const { quantity_change, movement_type, reason } = req.body;

    if (quantity_change === undefined || !movement_type) {
      return res.status(400).json({
        success: false,
        message: 'Quantity change and movement type are required'
      });
    }

    const result = await inventoryModel.update_stock(
      parseInt(id),
      parseInt(quantity_change),
      movement_type,
      reason || '',
      req.user.user_id
    );

    if (result.success) {
      res.json({
        success: true,
        message: 'Stock updated successfully',
        data: {
          previous_quantity: result.previous_quantity,
          new_quantity: result.new_quantity,
          quantity_change: result.quantity_change
        }
      });
    } else {
      res.status(400).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Update stock error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to update stock'
    });
  }
});

// Search products
router.get('/search', authenticateUser, async (req, res) => {
  try {
    const { q } = req.query;

    if (!q) {
      return res.status(400).json({
        success: false,
        message: 'Search query is required'
      });
    }

    const result = await inventoryModel.search_products(q);

    if (result.success) {
      res.json({
        success: true,
        data: result.products
      });
    } else {
      res.status(500).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Search products error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to search products'
    });
  }
});

// Get low stock alerts
router.get('/alerts/low-stock', authenticateUser, async (req, res) => {
  try {
    const result = await inventoryModel.get_low_stock_alerts();

    if (result.success) {
      res.json({
        success: true,
        data: result.alerts
      });
    } else {
      res.status(500).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Get low stock alerts error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to get low stock alerts'
    });
  }
});

// Generate barcode for product
router.post('/products/:id/barcode', authenticateUser, async (req, res) => {
  try {
    const { id } = req.params;
    const { barcode_type } = req.body;

    // Get product data
    const products_result = await inventoryModel.get_products();
    
    if (!products_result.success) {
      return res.status(500).json({
        success: false,
        message: products_result.error
      });
    }

    const product = products_result.products.find(p => p.id === parseInt(id));
    
    if (!product) {
      return res.status(404).json({
        success: false,
        message: 'Product not found'
      });
    }

    const result = await barcodeGenerator.generate_product_barcode(product);

    if (result.success) {
      res.json({
        success: true,
        data: {
          barcode_data: result.barcode_data,
          barcode_type: result.barcode_type,
          barcode_image: result.barcode_image,
          product_info: result.product_info
        }
      });
    } else {
      res.status(400).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Generate barcode error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to generate barcode'
    });
  }
});

// Generate batch barcodes
router.post('/barcodes/batch', authenticateUser, async (req, res) => {
  try {
    const { product_ids, barcode_type } = req.body;

    if (!product_ids || !Array.isArray(product_ids)) {
      return res.status(400).json({
        success: false,
        message: 'Product IDs array is required'
      });
    }

    // Get products data
    const products_result = await inventoryModel.get_products();
    
    if (!products_result.success) {
      return res.status(500).json({
        success: false,
        message: products_result.error
      });
    }

    const products = products_result.products.filter(p => product_ids.includes(p.id));

    const result = await barcodeGenerator.generate_batch_barcodes(products);

    if (result.success) {
      res.json({
        success: true,
        data: {
          generated_count: result.generated_count,
          failed_count: result.failed_count,
          barcodes: result.barcodes,
          failed_products: result.failed_products
        }
      });
    } else {
      res.status(400).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Generate batch barcodes error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to generate batch barcodes'
    });
  }
});

// Create barcode sheet
router.post('/barcodes/sheet', authenticateUser, async (req, res) => {
  try {
    const { barcodes, labels_per_row, labels_per_column } = req.body;

    if (!barcodes || !Array.isArray(barcodes)) {
      return res.status(400).json({
        success: false,
        message: 'Barcodes array is required'
      });
    }

    const result = await barcodeGenerator.create_barcode_sheet(
      barcodes,
      labels_per_row || 3,
      labels_per_column || 10
    );

    if (result.success) {
      res.json({
        success: true,
        data: {
          sheet_image: result.sheet_image,
          labels_count: result.labels_count,
          sheet_dimensions: result.sheet_dimensions
        }
      });
    } else {
      res.status(400).json({
        success: false,
        message: result.error
      });
    }

  } catch (error) {
    console.error('Create barcode sheet error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to create barcode sheet'
    });
  }
});

// Print barcode
router.post('/barcodes/:barcodeId/print', authenticateUser, async (req, res) => {
  try {
    const { barcodeId } = req.params;
    const { printer_name } = req.body;

    // This would integrate with printer service
    // For now, return success
    res.json({
      success: true,
      message: 'Barcode sent to printer successfully'
    });

  } catch (error) {
    console.error('Print barcode error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to print barcode'
    });
  }
});

// Get categories
router.get('/categories', authenticateUser, async (req, res) => {
  try {
    // This would require additional implementation in the inventory model
    // For now, return a placeholder response
    res.json({
      success: true,
      data: [
        { id: 1, name: 'Electronics', description: 'Electronic devices and accessories', color_code: '#00D4FF' },
        { id: 2, name: 'Clothing', description: 'Apparel and fashion items', color_code: '#00FFAB' },
        { id: 3, name: 'Food & Beverage', description: 'Food and drink items', color_code: '#FFD700' },
        { id: 4, name: 'Accessories', description: 'Miscellaneous accessories', color_code: '#E8E8E8' }
      ]
    });

  } catch (error) {
    console.error('Get categories error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to get categories'
    });
  }
});

// Get suppliers
router.get('/suppliers', authenticateUser, async (req, res) => {
  try {
    // This would require additional implementation in the inventory model
    // For now, return a placeholder response
    res.json({
      success: true,
      data: [
        { id: 1, name: 'ABC Electronics', contact_person: 'John Smith', email: 'john@abcelectronics.com', phone: '(555) 123-4567' },
        { id: 2, name: 'Fashion Forward', contact_person: 'Jane Doe', email: 'jane@fashionforward.com', phone: '(555) 234-5678' }
      ]
    });

  } catch (error) {
    console.error('Get suppliers error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to get suppliers'
    });
  }
});

// Get inventory statistics
router.get('/stats', authenticateUser, async (req, res) => {
  try {
    // This would require additional implementation in the inventory model
    // For now, return a placeholder response
    res.json({
      success: true,
      data: {
        total_products: 0,
        low_stock_items: 0,
        out_of_stock_items: 0,
        total_value: 0,
        categories_count: 4,
        suppliers_count: 2
      }
    });

  } catch (error) {
    console.error('Get inventory stats error:', error);
    res.status(500).json({
      success: false,
      message: 'Failed to get inventory statistics'
    });
  }
});

module.exports = router;